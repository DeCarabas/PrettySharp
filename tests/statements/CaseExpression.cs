class Foo {
    void Bar() {
        switch (isolated)
        {
        case TerminatorState.IsNamespaceMemberStartOrStop:
            if (this.IsNamespaceMemberStartOrStop())
            {
                return true;
            }

            break;
        case TerminatorState.IsAttributeDeclarationTerminator:
            if (this.IsAttributeDeclarationTerminator())
            {
                return true;
            }

            break;
        }
    }

    private CSharpSyntaxNode CheckRecursivePatternFeature(CSharpSyntaxNode node)
    {
        switch (node.Kind)
        {
        case SyntaxKind.RecursivePattern:
        case SyntaxKind.DiscardPattern:
        case SyntaxKind.VarPattern when ((VarPatternSyntax)node).Designation.Kind == SyntaxKind.ParenthesizedVariableDesignation:
            return this.CheckFeatureAvailability(node, MessageID.IDS_FeatureRecursivePatterns);

            // Haha this too!
        case DeclarationModifiers.Protected | DeclarationModifiers.Internal:
            // the two keywords "protected" and "internal" together are treated as one modifier.
            result &= ~DeclarationModifiers.AccessibilityMask;
            result |= DeclarationModifiers.ProtectedInternal;
            break;

        default:
            return node;
        }
    }
}
