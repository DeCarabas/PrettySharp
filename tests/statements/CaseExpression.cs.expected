class Foo
{
    void Bar()
    {
        switch (isolated)
        {
        case TerminatorState.IsNamespaceMemberStartOrStop:
            if (this.IsNamespaceMemberStartOrStop())
            {
                return true;
            }

            break;

        case TerminatorState.IsAttributeDeclarationTerminator:
            if (this.IsAttributeDeclarationTerminator())
            {
                return true;
            }

            break;
        }
    }

    private CSharpSyntaxNode CheckRecursivePatternFeature(CSharpSyntaxNode node)
    {
        switch (node.Kind)
        {
        case SyntaxKind.RecursivePattern:
        case SyntaxKind.DiscardPattern:
        case SyntaxKind.VarPattern when
                ((VarPatternSyntax)node).Designation.Kind ==
                SyntaxKind.ParenthesizedVariableDesignation:
            return this.CheckFeatureAvailability(
                node,
                MessageID.IDS_FeatureRecursivePatterns
            );

        default:
            return node;
        }
    }
}
